torch/csrc/quantized/quantized_backward.cpp

```diff
@@ -0,0 +1,76 @@
+#include <ATen/native/quantized/PackedParams.h>
+#include <torch/library.h>
+#include <torch/torch.h>
+
+namespace {
+using namespace torch::autograd;
+using namespace at;
+
+class PackedLinearWeightDynamicBackward :
+public Function<PackedLinearWeightDynamicBackward> {
+ public:
+  // Note that both forward and backward are static functions
+
+  // bias is an optional argument
+  static torch::Tensor forward(
+    AutogradContext *ctx,
+    at::Tensor input,
+    const c10::intrusive_ptr<LinearPackedParamsBase>& packed_weight,
+    bool reduce_range) {
+
+    static auto op =
+        at::Dispatcher::singleton()
+            .findSchemaOrThrow("quantized::linear_dynamic", "")
+            .typed<at::Tensor(
+              at::Tensor,
+              c10::intrusive_ptr<LinearPackedParamsBase,
+              c10::detail::intrusive_target_default_null_type<LinearPackedParamsBase>> const&,
+              bool)>();
+    auto output = op.redispatch(
+      DispatchKeySet({DispatchKey::CPU}), input, packed_weight, reduce_range);
+    auto [weight, bias] = packed_weight->unpack();
+    TORCH_CHECK(weight.qscheme() == c10::kPerTensorAffine);
+    // TO-DO : passing a bias does not work, hence create an arbitrary one in backward loop
+    ctx->save_for_backward({weight});
+    return output;
+  }
+
+  static tensor_list backward(AutogradContext* ctx, tensor_list grad_outputs) {
+    auto saved = ctx->get_saved_variables();
+    auto original_weight = saved[0];
+    original_weight = at::permute(original_weight, {1, 0});
+    auto grad_output = grad_outputs[0];
+    static auto op =
+        at::Dispatcher::singleton()
+            .findSchemaOrThrow("quantized::linear_prepack", "")
+            .typed<c10::intrusive_ptr<LinearPackedParamsBase>(
+              at::Tensor,
+              c10::optional<at::Tensor>)>();
+    c10::optional<at::Tensor> bias;
+    auto prepacked_weight = op.redispatch(
+      DispatchKeySet({DispatchKey::QuantizedCPU}), original_weight, bias);
+    auto grad_input = prepacked_weight->apply_dynamic(grad_output);
+    return {grad_input, torch::Tensor(), torch::Tensor()};
+  }
+};
+
+at::Tensor packed_linear_weight_grad(
+    at::Tensor input,
```

Comment:
This should take `c10::DispatchKeySet ks, ` as the first argument. And this keyset should be used when redispatching during the forward pass. similar to https://github.com/pytorch/pytorch/blob/7a192cc51c7172860efd35413acf9a8b9aafd2c9/torch/csrc/autograd/VariableTypeManual.cpp#L110-L122